### ISSUE TITLE
Detect and fail properly for invalid files

### ISSUE THREAD

**Issue by EnTeQuAk** [mozilla/addons-linter#556]
Example error output:

```bash
$ addons-linter ./src/olympia/devhub/tests/addons/unopenable.xpi
end of central directory record signature not found
```

The file isn't an actual zip: https://github.com/mozilla/addons-server/blob/master/src/olympia/devhub/tests/addons/unopenable.xpi

Expected behavior:

```json
"errors": [{
    "_type": "error",
    "code": "BAD_ZIPFILE",
    "message" : "Corrupt ZIP file",
    "description" : "We were unable to decompress the zip file.",
    "file": null
}]
```

the `code`, `message` and `description` can be anythhing, that's just the error message that `amo-validator` has.
